index.ts

```diff
@@ -0,0 +1,88 @@
+import { EventEmitter } from 'events';
+
+type Handler = (...args: any[]) => void;
+type EventMap = {
+  [k: string]: Handler | Handler[] | undefined;
+};
+
+/* istanbul ignore next */
+function _ReflectApply<T, A extends any[], R> (target: (this: T, ...args: A) => R, receiver: T, args: A): R {
+  return Function.prototype.apply.call(target, receiver, args);
+}
+
+const R = typeof Reflect === 'object'
+  ? Reflect
+  /* istanbul ignore next */
+  : null;
+
+const ReflectApply = R && typeof R.apply === 'function'
+  ? R.apply
+  /* istanbul ignore next */
+  : _ReflectApply;
+
```

Comment:
I think `_ReflectApply` can be removed. It appears to be a shim, and it's not impossible that this code was originally written before `Reflect` was implemented everywhere, including Node. Today, however, I'm not sure what environment wouldn't be able to run it: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect

Reply:
Hmm, that's a compelling argument. I was generally avoiding changes to the logic but since we are going to be releasing this in a major version bump _anyway_... this seems pretty safe.

<img width="1206" alt="Browser compatibility table for Reflect, from MDN" src="https://user-images.githubusercontent.com/1623628/73954602-9e0eda80-48dc-11ea-9208-6858438f5fe0.png">

This table does seem quite green.

Reply:
I'll light a candle for our Internet Explorer users. 😄